### ISSUE TITLE
vulnxscan fails if the Nix output path is a JSON file

### ISSUE THREAD

**Issue by nlewo** [tiiuae/sbomnix#56]
First of all, thanks for you project!

The [nix2container](https://github.com/nlewo/nix2container) project produces container image specification as JSON file. When running `vulnxscan` on a nix2container output path, it fails such as:
```
$ nix run github:tiiuae/sbomnix#vulnxscan -- ./result
CRITICAL Specified target is not a nix artifact: 'result'

$ file ./result
./result: symbolic link to /nix/store/7s4y6dcmfc6frqv38j8y6g7ifmazh5hx-image-bash.json
```

It seems to be because vulnxscan only consider non JSON file as Nix artifact: https://github.com/tiiuae/sbomnix/blob/f763f71670bfcd7fb4f6a480a7e7cc56e53336c6/scripts/vulnxscan/vulnxscan.py#L338

Instead of testing the file type, maybe you should run a Nix command on this store path to let Nix deciding if it is a Nix artefact or not. (`nix-store -q <FILE>` for instance).

Note this is related to this [nix2container issue](https://github.com/nlewo/nix2container/issues/70#issuecomment-1517390681).

**Comment by henrirosten** [comment]
Thanks for the bug report @nlewo!

Can you verify if https://github.com/tiiuae/sbomnix/pull/57 fixes the issue?

**Comment by nlewo** [comment]
Tested and lgtm. Thx.

**Comment by henrirosten** [comment]
Thanks, the fix is no merged to main: c8d02ef59ca227b6fabd9429e5e17f3d7ca808d0
